### PR TITLE
feat: a3 game select and save variables

### LINKED ISSUE #127: 🚀 Feature: a3 store variables in game context
blocks a ton of stuff
* [ ] :wave: I may be able to implement this feature request

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by reginawang99**
closes #127 

unblocks #112

### CODE REVIEW

**Review by darren2hang — COMMENTED** [review]
This looks really good! I have a one question though. Why did you decide to use a switch statement instead of a dictionary/hashmap? My intuition tells the dictionary would have faster runtime, but I don't think it matters much in this case, since the map is only like 6 long.

**Comment by reginawang99** [comment]
oooo good point, in my brain, the switch statement is easier to read (cuz mapping + return value in one area)! tbh im down for either and am down to change, but am v curious as to what @BryanPan342 thinks also

File: src/components/Activities/Activity3/Game/index.tsx [hunk 1]
```diff
@@ -2,24 +2,27 @@ import React, { useContext, useEffect, useState } from 'react';
...
 
-export const GameContext = React.createContext({
-  setState: (_state: A3_GAME_STATE): void => undefined,
-  goNextState: (): void => undefined,
-});
+export const GameContext = React.createContext<Partial<{
+  setState: (state: A3_GAME_STATE) => void,
+  goNextState: () => void,
+  variableSelection: VARIABLES[],
+}>>({});
```
Comment (BryanPan342) [inline src/components/Activities/Activity3/Game/index.tsx:12]: make the Partial an interface!

```
interface IGameContext {
  setState: (state: A3_GAME_STATE) => void,
  goNextState: () => void,
  variableSelection: VARIABLES[],
}
```

**Review by BryanPan342 — CHANGES_REQUESTED** [review]
minor nit

**Review by BryanPan342 — APPROVED** [review]
LGTM

**Pull request merged**